Added new table that shows RA for swab samples.

simonleandergrimm opened the pull request:
To support the table and the addition of Zephyr12b, I did the following:
 - As I'm now at Zephyr12b, I need to switch metadata parsing from mgs-restricted to mgs-metadata.
    - Added parsing of read_counts.tsv.gz.
 - Support identifying duplicates in non-BLASTed swab reads (e.g., SARS-CoV-2) using minimap2 alignments.

@jeffkaufman Let me know if I should factor out the read count parsing, currently that's tacked onto `prep-*-validation.py`, which isn't very clean.

jeffkaufman commented on scripts/prep-swab-validation.py:
```diff
@@ -24,43 +24,82 @@
     "KY766069.1",  # Zika virus isolate
 ]
 
+WIGGLE_ROOM=3 # count dups even if start/end is off by this many bp
```
FWIW I'm not sure this is a reasonable way to dedup ONT data.  This shouldn't block this PR, but it would be good to check in with Katherine or do some reading about what's recommended.

simonleandergrimm replied:
Created a thread here: https://twist.com/a/197793/ch/619193/t/6904068/

jeffkaufman commented on scripts/prep-swab-validation.py:
```diff
@@ -24,43 +24,82 @@
+    if delivery == "NAO-ONT-20250328-Zephyr12b":
+        version = "v2.8.3.1"
+    else:
+        version = "v2.8.1.3-dev"
```
This is not great -- this means that if there are important differences in the pipeline then we're changing our handling partway through.  We really should use a consistent version of the pipeline for all deliveries we're analyzing together.  Options are rerunning 12b with `v2.8.1.3-dev` at the same commit you used for the other analysis, or rerunning the others at the released `v2.8.3`.

simonleandergrimm replied:
I'll rerun everything so it's on version v2.8.3.2. Hope fully shouldn't take too long, I just did it for one delivery and it only reran the FASTQC step.

simonleandergrimm replied:
This is now done, updted the version in the script.

jeffkaufman commented on scripts/swab-ra-table.py:
```diff
@@ -0,0 +1,135 @@
+# Load data
+taxid_names = load_taxonomy_names()
+
+with open(os.path.join(dashboard_dir, "metadata_samples.json")) as f:
```
I thought this didn't work now that you had Zephyr12b?

simonleandergrimm replied:
This was left over, just deleted it.

jeffkaufman commented on scripts/swab-ra-table.py:
```diff
@@ -0,0 +1,135 @@
+from datetime import datetime
+
+# Add local module path
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
```
Do you really need this?  I'd expect local imports to just work

simonleandergrimm replied:
Right, just fixed this.

jeffkaufman commented on scripts/swab-ra-table.py:
```diff
@@ -0,0 +1,135 @@
+
+
+# Initialize data structures
+samples = defaultdict(lambda: defaultdict(int))  # (date, location, pathogen) -> counts
```
defaultdict(Counter)

simonleandergrimm commented:
@jeffkaufman I updated [scripts/swab-ra-table.py](https://github.com/naobservatory/swab-based-p2ra/pull/3/files/ba48fad34a3cca8c6c95d6600e1fde64eda4a7a1#diff-058592ed695f1748cbc68f7d1d7e998979ab99ff5e88aa94a38e9a1841c2aed2) to fix the faulty read number issue. I also added a second table that returns HV and read number information broken down by day/location/protocol.

jeffkaufman commented on scripts/swab-ra-table.py:
```diff
@@ -4,36 +4,63 @@
 
 
+# Create per-category counts and get treatment for each sample
+date_loc_read_counts = defaultdict(int)
```
nit: use `Counter()` instead of `defaultdict(int)`

simonleandergrimm replied:
Done.

jeffkaufman commented:
> @jeffkaufman I updated [scripts/swab-ra-table.py](https://github.com/naobservatory/swab-based-p2ra/pull/3/files/ba48fad34a3cca8c6c95d6600e1fde64eda4a7a1#diff-058592ed695f1748cbc68f7d1d7e998979ab99ff5e88aa94a38e9a1841c2aed2) to fix the faulty read number issue. I also added a second table that returns HV and read number information broken down by day/location/protocol.

Next time can you also assign to me so I'll see in  https://github.com/issues/assigned?q=is%3Apr%20state%3Aopen%20archived%3Afalse%20assignee%3A%40me%20sort%3Aupdated-desc%20

simonleandergrimm commented:
> Next time can you also assign to me so I'll see in https://github.com/issues/assigned?q=is%3Apr%20state%3Aopen%20archived%3Afalse%20assignee%3A%40me%20sort%3Aupdated-desc%20
Will do!
